pkg/sync/sync_context.go

```diff
@@ -707,10 +707,12 @@ func (sc *syncContext) runTasks(tasks syncTasks, dryRun bool) runState {
 	sc.log.WithFields(log.Fields{"numTasks": len(tasks), "dryRun": dryRun}).Debug("running tasks")
 
 	runState := successful
+	var runStateMutex sync.Mutex
 	var createTasks syncTasks
 	var pruneTasks syncTasks
 
 	for _, task := range tasks {
+		task := task
```

Comment:
I wonder why is this needed?

Reply:
I think that's been fixed with this commit.
https://github.com/argoproj/gitops-engine/pull/51/commits/c5d445f642a1283a09aa0c0056274e1e6fe17997